src/lib/swapper/swappers/OneInchSwapper/executeTrade/executeTrade.ts

```diff
@@ -57,13 +58,18 @@ export async function executeTrade(
       })
     }
 
+    const { average } = await adapter.getGasFeeData()
```

Comment:
this whole helper function should be updated to use the same swapper helpers. we also can remove the asset validation logic here and add [assertValidTradePair](https://github.com/shapeshift/web/blob/2287f6e7fcbce594e425d8467345255573373ff5/src/lib/swapper/swappers/ZrxSwapper/utils/helpers/helpers.ts#L91) to the `getQuote` helper.

Reply:
love this idea, do you reckon we could do it as a follow-up task to minimise the scope of this PR?

Reply:
For sure. I'm going to be in there for the remaining gas oracle work, so I will hit that in my pr :)

Reply:
gm